alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultAuthorizationUtility.java

```diff
@@ -0,0 +1,136 @@
+        for (final Long roleId : roleIds) {
+            final Optional<String> roleName = getRoleName(roleId);
+            roleName.ifPresent(role -> userRoles.add(UserRoleModel.of(role, readPermissionsForRole(roleId))));
+        }
+        return userRoles;
+    }
+
+    @Override
+    public void updateUserRoles(final Long userId, final Collection<UserRoleModel> roles) {
+        if (null != userId) {
+            userRoleRepository.deleteAllByUserId(userId);
+
+            if (null != roles && !roles.isEmpty()) {
+                final Collection<String> roleNames = roles.stream().map(UserRoleModel::getName).collect(Collectors.toSet());
+                final List<RoleEntity> roleEntities = roleRepository.findRoleEntitiesByRoleNames(roleNames);
+                final List<UserRoleRelation> roleRelations = new LinkedList<>();
+                for (final RoleEntity role : roleEntities) {
+                    roleRelations.add(new UserRoleRelation(userId, role.getId()));
+                }
+                userRoleRepository.saveAll(roleRelations);
+            }
+        }
+    }
+
+    private Optional<String> getRoleName(final Long roleId) {
+        return roleRepository.findById(roleId).map(RoleEntity::getRoleName);
+    }
+
+    @Override
+    public PermissionMatrixModel mergePermissionsForRoles(final Collection<String> roleNames) {
+        final List<RoleEntity> roles = roleRepository.findRoleEntitiesByRoleNames(roleNames);
+        final List<Long> roleIds = roles.stream().map(RoleEntity::getId).collect(Collectors.toList());
+        final List<PermissionMatrixRelation> permissions = permissionMatrixRepository.findAllByRoleIdIn(roleIds);
+        return readPermissionsForRole(permissions);
+    }
+
+    @Override
+    public PermissionMatrixModel readPermissionsForRole(final Long roleId) {
+        final List<PermissionMatrixRelation> permissions = permissionMatrixRepository.findAllByRoleId(roleId);
+        return readPermissionsForRole(permissions);
+    }
+
+    private PermissionMatrixModel readPermissionsForRole(final List<PermissionMatrixRelation> permissions) {
+        final Map<String, EnumSet<AccessOperation>> permissionOperations = new HashMap<>();
+
+        if (null != permissions) {
+            for (final PermissionMatrixRelation relation : permissions) {
+                final Optional<PermissionKeyEntity> permissionKey = permissionKeyRepository.findById(relation.getPermissionKeyId());
+                final Optional<AccessOperationEntity> accessOperation = accessOperationRepository.findById(relation.getAccessOperationId());
+                permissionKey.ifPresent(key -> {
+                    final String keyName = key.getKeyName();
+                    permissionOperations.computeIfAbsent(keyName, ignored -> EnumSet.noneOf(AccessOperation.class));
+                    accessOperation.ifPresent(operation -> permissionOperations.get(keyName).add(AccessOperation.valueOf(operation.getOperationName())));
```

Comment:
accessOperation and permissionKey are both created and then only checked if they're present. We can probably combine them into a single line.